src/main/java/io/cryostat/configuration/ConfigurationModule.java

```diff
@@ -77,36 +70,24 @@ static Path provideConfigurationPath(Logger logger, Environment env) {
     @Provides
     @Singleton
     static CredentialsManager provideCredentialsManager(
-            @Named(CONFIGURATION_PATH) Path confDir,
             MatchExpressionValidator matchExpressionValidator,
             MatchExpressionEvaluator matchExpressionEvaluator,
-            FileSystem fs,
-            DiscoveryStorage storage,
+            DiscoveryStorage discovery,
+            StoredCredentialsDao dao,
             NotificationFactory notificationFactory,
-            Gson gson,
             Logger logger) {
-        try {
-            Path credentialsDir = confDir.resolve(CREDENTIALS_SUBDIRECTORY);
-            if (!fs.isDirectory(credentialsDir)) {
-                Files.createDirectory(
-                        credentialsDir,
-                        PosixFilePermissions.asFileAttribute(
-                                Set.of(
-                                        PosixFilePermission.OWNER_READ,
-                                        PosixFilePermission.OWNER_WRITE,
-                                        PosixFilePermission.OWNER_EXECUTE)));
-            }
-            return new CredentialsManager(
-                    credentialsDir,
-                    matchExpressionValidator,
-                    matchExpressionEvaluator,
-                    fs,
-                    storage,
-                    notificationFactory,
-                    gson,
-                    logger);
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
+        return new CredentialsManager(
+                matchExpressionValidator,
+                matchExpressionEvaluator,
+                discovery,
+                dao,
+                notificationFactory,
+                logger);
+    }
+
+    @Provides
+    @Singleton
+    static StoredCredentialsDao providePluginInfoDao(EntityManager em, Logger logger) {
```

Comment:
Should be `provideStoredCredentialsDao`?

Reply:
Thanks, fixed.